inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/querybuilder/SPARQLQueryBuilder.java

```diff
@@ -1556,6 +1568,104 @@ public boolean exists(RepositoryConnection aConnection, boolean aAll)
         return result;    
     }
     
+    @Override
+    public List<KBStatement> asStatements(RepositoryConnection aConnection, boolean aAll)
+    {
+        long startTime = currentTimeMillis();
+        String queryId = toHexString(hashCode());
+        
+        projections.add(VAR_PREDICATE);
+        projections.add(VAR_OBJECT);
+        addPattern(PRIMARY, VAR_SUBJECT.has(VAR_PREDICATE, VAR_OBJECT));
+
+        String queryString = selectQuery().getQueryString();
+        LOG.trace("[{}] Query: {}", queryId, queryString);
+
+        List<KBStatement> results;
+        if (returnEmptyResult) {
+            results = emptyList();
+            
+            LOG.debug("[{}] Query was skipped because it would not return any results anyway",
+                    queryId);
```

Comment:
You could just return here instead of the long if indented block

Reply:
Changed.